From fd1cfbb38f97c9dd6ce9fe539d143dcfdac7fdad Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Mon, 16 Mar 2015 22:43:13 -0400
Subject: [PATCH] Force update player velocity


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index a75b86ca..bfa93c05 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1327,6 +1327,13 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
     // Paper - Undeprecate
     public void resetTitle();
 
+    /**
+     * Immediately send a velocity update packet to the player, after firing a {@link org.bukkit.event.player.PlayerVelocityEvent}.
+     * This should be called immediately after {@link #setVelocity} to ensure the client receives the velocity unaltered.
+     * Otherwise, it may be affected by ground friction before being sent.
+     */
+    public void updateVelocity();
+
     // Spigot start
     public class Spigot extends Entity.Spigot
     {
-- 
2.50.1 (Apple Git-155)

