From d8feca71723778ab4dfb3f6d1e6f0e0f0b589ce7 Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Tue, 17 Mar 2015 03:02:27 -0400
Subject: [PATCH] Relative teleport API


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index bfa93c05..2156211f 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -1334,6 +1334,14 @@ public interface Player extends HumanEntity, Conversable, CommandSender, Offline
      */
     public void updateVelocity();
 
+    /**
+     * Seamlessly move the player by the given offset and rotation angles.
+     * Because the relative calculation is applied by the client, you cannot
+     * know the player's absolute location immediately after teleporting.
+     */
+    public boolean teleportRelative(org.bukkit.util.Vector deltaPos, float deltaYaw, float deltaPitch, org.bukkit.event.player.PlayerTeleportEvent.TeleportCause cause);
+    public boolean teleportRelative(org.bukkit.util.Vector deltaPos, float deltaYaw, float deltaPitch);
+
     // Spigot start
     public class Spigot extends Entity.Spigot
     {
-- 
2.50.1 (Apple Git-155)

