From 7dbcd6ccddbd4f8d595bedfbc0482695b0a2d739 Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Sat, 20 Jun 2015 12:27:57 -0400
Subject: [PATCH] Call onDisable before disabling the plugin


diff --git a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
index c961b18c..12d797f4 100644
--- a/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
+++ b/src/main/java/org/bukkit/plugin/java/JavaPlugin.java
@@ -314,13 +314,19 @@ public abstract class JavaPlugin extends PluginBase {
      * @param enabled true if enabled, otherwise false
      */
     protected final void setEnabled(final boolean enabled) {
-        if (isEnabled != enabled) {
-            isEnabled = enabled;
-
-            if (isEnabled) {
+        if(!isEnabled && enabled) {
+            isEnabled = true;
+            try {
                 onEnable();
-            } else {
+            } catch(RuntimeException e) {
+                isEnabled = false;
+                throw e;
+            }
+        } else if(isEnabled && !enabled) {
+            try {
                 onDisable();
+            } finally {
+                isEnabled = false;
             }
         }
     }
-- 
2.50.1 (Apple Git-155)

