From ee5ee1ec958a04bc4a443c9614d5243804c06963 Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Sun, 6 Mar 2016 03:43:35 -0500
Subject: [PATCH] Empty chunk API


diff --git a/src/main/java/org/bukkit/Chunk.java b/src/main/java/org/bukkit/Chunk.java
index fce2c50a..65e1a0e7 100644
--- a/src/main/java/org/bukkit/Chunk.java
+++ b/src/main/java/org/bukkit/Chunk.java
@@ -51,6 +51,11 @@ public interface Chunk extends Physical {
      */
     Set<Block> getBlocks(Material material);
 
+    /**
+     * Is this chunk entirely filled with air blocks?
+     */
+    boolean isEmpty();
+
     /**
      * Capture thread-safe read-only snapshot of chunk data
      *
-- 
2.50.1 (Apple Git-155)

