From 72bbea9684c3ea80518423bb03f2aa89ceb2b94d Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Wed, 7 Dec 2016 04:29:15 -0500
Subject: [PATCH] FallingBlock form block flag


diff --git a/src/main/java/org/bukkit/entity/FallingBlock.java b/src/main/java/org/bukkit/entity/FallingBlock.java
index 240a2783..3bb6f16e 100644
--- a/src/main/java/org/bukkit/entity/FallingBlock.java
+++ b/src/main/java/org/bukkit/entity/FallingBlock.java
@@ -66,4 +66,20 @@ public interface FallingBlock extends Entity {
      * @return the source block location the falling block was spawned from
      */
     org.bukkit.Location getSourceLoc(); // PaperSpigot - Add FallingBlock source location API
+
+    /**
+     * Can this entity form a block when it lands?
+     */
+    boolean canFormBlock();
+
+    /**
+     * Allow or prevent this entity from forming a block when it lands.
+     *
+     * If this is false, the entity will just disappear when it lands.
+     * If true, it may form a block if other circumstances permit it.
+     *
+     * This is initially true by default.
+     */
+    void setFormBlock(boolean formBlock);
+
 }
-- 
2.50.1 (Apple Git-155)

