From cec8ae12cd5f5a01be36e34a93109de93f76c2b4 Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Fri, 24 Aug 2018 21:34:40 -0700
Subject: [PATCH] Trade API


diff --git a/src/main/java/org/bukkit/entity/HumanEntity.java b/src/main/java/org/bukkit/entity/HumanEntity.java
index 053e61d..8112f5c 100644
--- a/src/main/java/org/bukkit/entity/HumanEntity.java
+++ b/src/main/java/org/bukkit/entity/HumanEntity.java
@@ -65,6 +65,14 @@ public interface HumanEntity extends LivingEntity, AnimalTamer, Permissible, Inv
     public InventoryView openInventory(Inventory inventory);
 
     /**
+     * Makes a copy of the given villager's current trading offers and opens
+     * a trading window for this player, allowing them to interact with those
+     * offers, without affecting the villager's state, or any other players who
+     * might be trading with the villager.
+     */
+    public InventoryView openMerchantCopy(Villager trader);
+
+    /**
      * Opens an empty workbench inventory window with the player's inventory
      * on the bottom.
      *
-- 
2.6.4

