From 045ebb616c56f4249d6b55130241894419989b23 Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Sat, 15 Sep 2018 01:43:42 -0700
Subject: [PATCH] Add Block Strength API


diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
index e617106..6a089e6 100644
--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -111,6 +111,23 @@ public interface Block extends Metadatable, Physical {
     byte getLightFromBlocks();
 
     /**
+     * Sets the block strength of this block back to the default value
+     */
+    void resetBlockStrength();
+
+    /**
+     * Set the strength of this Block
+     */
+    void setBlockStrength(float blockStrength);
+
+    /**
+     * Gets the strength of this block
+     *
+     * @return Float representing the strength of this block
+     */
+    float getBlockStrength();
+
+    /**
      * Gets the world which contains this Block
      *
      * @return World containing this block
-- 
2.6.4

