From 15111100ff114a3e8e821a49c0ae7c69ba19169e Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Sat, 4 Aug 2018 17:35:24 -0700
Subject: [PATCH] Don't filter negative damage values


diff --git a/src/main/java/net/minecraft/server/ItemStack.java b/src/main/java/net/minecraft/server/ItemStack.java
index 015ce283..6a365252 100644
--- a/src/main/java/net/minecraft/server/ItemStack.java
+++ b/src/main/java/net/minecraft/server/ItemStack.java
@@ -339,11 +339,6 @@ public final class ItemStack {
             i = 0;
         }
         // CraftBukkit end
-        this.damage = i;
-        if (this.damage < -1) { // CraftBukkit
-            this.damage = 0;
-        }
-
     }
 
     public int j() {
-- 
2.18.0

