From 8002d55dcecce2a4db48e721db291d8e7c4d9d8b Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Tue, 3 Jun 2014 08:02:58 -0400
Subject: [PATCH] Fix head rotation packet spam


diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 1fc2388ea..12112ce7a 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -411,8 +411,10 @@ public class EntityTrackerEntry {
                     }
 
                     // CraftBukkit start - Fix for nonsensical head yaw
-                    this.i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
-                    this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                    if(this.tracker instanceof EntityLiving) {
+                        this.i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
+                        this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                    }
                     // CraftBukkit end
 
                     if (this.tracker instanceof EntityLiving) {
-- 
2.50.1 (Apple Git-155)

