From 2b9879d2938eec7fef8dab7a48109bd6c7ad6568 Mon Sep 17 00:00:00 2001
From: Jedediah Smith <jedediah@silencegreys.com>
Date: Mon, 1 Jun 2015 06:29:07 -0400
Subject: [PATCH] Improve BlockState API


diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockState.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockState.java
index 4a8c8141b..8a2a578af 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockState.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockState.java
@@ -83,19 +83,32 @@ public class CraftBlockState implements BlockState {
         return chunk;
     }
 
-    public void setData(final MaterialData data) {
-        Material mat = getType();
+    @Override
+    public Material getMaterial() {
+        return Material.getMaterial(getTypeId());
+    }
 
-        if ((mat == null) || (mat.getData() == null)) {
-            this.data = data;
-        } else {
-            if ((data.getClass() == mat.getData()) || (data.getClass() == MaterialData.class)) {
-                this.data = data;
-            } else {
-                throw new IllegalArgumentException("Provided data is not of type "
-                        + mat.getData().getName() + ", found " + data.getClass().getName());
-            }
+    @Override
+    public MaterialData getMaterialData() {
+        return data;
+    }
+
+    @Override
+    public void setMaterial(Material material) {
+        setTypeId(material.getId());
+    }
+
+    @Override
+    public void setMaterialData(MaterialData materialData) {
+        this.type = materialData.getItemTypeId();
+        this.data = materialData;
+    }
+
+    public void setData(final MaterialData data) {
+        if(getType() != data.getItemType()) {
+            throw new IllegalArgumentException("Provided data has wrong material " + data.getItemType() + ", must be " + getType());
         }
+        this.data = data;
     }
 
     public MaterialData getData() {
-- 
2.50.1 (Apple Git-155)

