From 5157b84a89effeb4cd4a60393afc0bcb48cf64f8 Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Sun, 12 Aug 2018 23:18:21 -0700
Subject: [PATCH] Add PlayerUseUnknownEntityEvent


diff --git a/src/main/java/net/minecraft/server/PacketPlayInUseEntity.java b/src/main/java/net/minecraft/server/PacketPlayInUseEntity.java
index a8fc3361a..d162ad6b2 100644
--- a/src/main/java/net/minecraft/server/PacketPlayInUseEntity.java
+++ b/src/main/java/net/minecraft/server/PacketPlayInUseEntity.java
@@ -4,7 +4,7 @@ import java.io.IOException;
 
 public class PacketPlayInUseEntity implements Packet<PacketListenerPlayIn> {
 
-    private int a;
+    private int a;public int getEntityId() { return this.a; } // Paper - add accessor
     private PacketPlayInUseEntity.EnumEntityUseAction action;
     private Vec3D c;
 
diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 18343f7c3..d23ff4023 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1504,6 +1504,11 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                 }
             }
         }
+        // Paper start - fire event
+        else {
+            new org.github.paperspigot.event.player.PlayerUseUnknownEntityEvent(this.getPlayer(), packetplayinuseentity.getEntityId(), packetplayinuseentity.a() == PacketPlayInUseEntity.EnumEntityUseAction.ATTACK).callEvent();
+        }
+        // Paper end
 
     }
 
-- 
2.18.0

