From a4de48d64f1d36014ae6942b8676aba2d5a48cdb Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Tue, 14 Aug 2018 00:29:18 -0700
Subject: [PATCH] Fix AIOOBE in inventory handling


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index d23ff4023..e9f9128f4 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1602,7 +1602,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                     }
                     if (packetplayinwindowclick.c() == 0 || packetplayinwindowclick.c() == 1) {
                         action = InventoryAction.NOTHING; // Don't want to repeat ourselves
-                        if (packetplayinwindowclick.b() == -999) {
+                        if (packetplayinwindowclick.b() < 0) { // Paper - GH-404
                             if (player.inventory.getCarried() != null) {
                                 action = packetplayinwindowclick.c() == 0 ? InventoryAction.DROP_ALL_CURSOR : InventoryAction.DROP_ONE_CURSOR;
                             }
-- 
2.18.0

