From d08ad68f1213c92be82f43cea062776778b79f7a Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Tue, 21 Aug 2018 16:10:45 -0700
Subject: [PATCH] Avoid item merge if stack size above max stack size


diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index 789d58332..c0e756c66 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -145,6 +145,10 @@ public class EntityItem extends Entity {
     // Spigot end
 
     private void w() {
+        // Paper start - avoid item merge if stack size above max stack size
+        ItemStack stack = getItemStack();
+        if (stack.count >= stack.getMaxStackSize()) return;
+        // Paper end
         // Spigot start
         double radius = world.spigotConfig.itemMerge;
         Iterator iterator = this.world.a(EntityItem.class, this.getBoundingBox().grow(radius, radius, radius)).iterator();
-- 
2.18.0

