From 83b48a73fc8cba0c3994f78ae3a16ea63d01642a Mon Sep 17 00:00:00 2001
From: cswhite2000 <18whitechristop@gmail.com>
Date: Fri, 24 Aug 2018 02:33:02 -0700
Subject: [PATCH] isChunkGenerated API


diff --git a/src/main/java/net/minecraft/server/ChunkProviderServer.java b/src/main/java/net/minecraft/server/ChunkProviderServer.java
index 475e2507d..0602beac9 100644
--- a/src/main/java/net/minecraft/server/ChunkProviderServer.java
+++ b/src/main/java/net/minecraft/server/ChunkProviderServer.java
@@ -129,6 +129,12 @@ public class ChunkProviderServer implements IChunkProvider {
 
     }
 
+    // SportPaper start
+    public boolean isChunkGenerated(int x, int z) {
+        return this.chunks.containsKey(ChunkCoordIntPair.asLong(x, z)) || ((ChunkRegionLoader)this.chunkLoader).chunkExists(world, x, z);
+    }
+    // SportPaper end
+
     // CraftBukkit start - Add async variant, provide compatibility
     public Chunk getChunkIfLoaded(int x, int z) {
         return chunks.get(LongHash.toLong(x, z));
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 27840e017..c70c4c174 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -663,6 +663,12 @@ public class CraftWorld implements World {
         return getChunkAt(location.getBlockX() >> 4, location.getBlockZ() >> 4);
     }
 
+    // Paper start
+    public boolean isChunkGenerated(int x, int z) {
+        return this.getHandle().chunkProviderServer.isChunkGenerated(x, z);
+    }
+    // Paper end
+
     public ChunkGenerator getGenerator() {
         return generator;
     }
-- 
2.30.2

