From b1df953e7bd0be5aa0ee8bfc93a9da426c64b3cc Mon Sep 17 00:00:00 2001
From: Austin Mayes <austin@avicus.net>
Date: Thu, 27 Sep 2018 08:31:55 -0500
Subject: [PATCH] Add translatable methods to world broadcast API


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 19835eac7..4135d5954 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -15,6 +15,7 @@ import java.util.UUID;
 import net.md_5.bungee.api.chat.BaseComponent;
 import net.minecraft.server.*;
 
+import network.stratus.sportpaper.api.text.TranslatableComponent;
 import org.apache.commons.lang.Validate;
 import org.bukkit.BlockChangeDelegate;
 import org.bukkit.Bukkit;
@@ -874,6 +875,16 @@ public class CraftWorld implements World {
         getPlayers().forEach(player -> player.sendMessage(components));
     }
 
+    @Override
+    public void broadcast(TranslatableComponent... translatables) {
+        getPlayers().forEach(player -> player.sendMessage(translatables));
+    }
+
+    @Override
+    public void broadcast(TranslatableComponent translatable) {
+        getPlayers().forEach(player -> player.sendMessage(translatable));
+    }
+
     public void save() {
     // Spigot start
         save(true);
-- 
2.30.2

