From 6fcc8bda02bc6cd0d7e27efba92817e21f66c00e Mon Sep 17 00:00:00 2001
From: Indicado <indicado@walrus.gg>
Date: Wed, 15 Jul 2020 09:52:19 +0200
Subject: [PATCH] Add option to remove randomness in arrow damage


diff --git a/src/main/java/net/minecraft/server/EntityArrow.java b/src/main/java/net/minecraft/server/EntityArrow.java
index 4366f34b..dd903aa7 100644
--- a/src/main/java/net/minecraft/server/EntityArrow.java
+++ b/src/main/java/net/minecraft/server/EntityArrow.java
@@ -247,7 +247,7 @@ public class EntityArrow extends Entity implements IProjectile {
                     f2 = MathHelper.sqrt(this.motX * this.motX + this.motY * this.motY + this.motZ * this.motZ);
                     int k = MathHelper.f((double) f2 * this.damage);
 
-                    if (this.isCritical()) {
+                    if (this.isCritical() && PaperSpigotConfig.includeRandomnessInArrowDamage) { // SportPaper
                         k += this.random.nextInt(k / 2 + 2);
                     }
 
diff --git a/src/main/java/org/github/paperspigot/PaperSpigotConfig.java b/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
index 038137dc..8d58d1f3 100644
--- a/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
+++ b/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
@@ -118,6 +118,11 @@ public class PaperSpigotConfig
         includeRandomnessInArrowTrajectory = getBoolean("settings.include-randomness-in-arrow-trajectory", includeRandomnessInArrowTrajectory);
     }
 
+    public static boolean includeRandomnessInArrowDamage = true;
+    private static void includeRandomnessInArrowDamage() {
+        includeRandomnessInArrowDamage = getBoolean("settings.include-randomness-in-arrow-damage", includeRandomnessInArrowDamage);
+    }
+
     public static boolean tickEmptyWorlds = true;
     private static void tickEmptyWorlds() {
         tickEmptyWorlds = getBoolean("settings.tick-empty-worlds", tickEmptyWorlds);
-- 
2.15.1.windows.2

