From a173bc41dec525e1bac364ba195ae33e5fb90abc Mon Sep 17 00:00:00 2001
From: linsaftw <linsaftw@users.noreply.github.com>
Date: Sun, 25 Apr 2021 17:33:56 -0300
Subject: [PATCH] Fix Teleport Invisibility


diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 45d611f34..97fbf0eab 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -13,6 +13,10 @@ import org.bukkit.entity.Player;
 import org.bukkit.event.player.PlayerVelocityEvent;
 // CraftBukkit end
 
+// SportPaper start
+import java.util.ArrayList;
+// SportPaper end
+
 public class EntityTrackerEntry {
 
     private static final Logger p = LogManager.getLogger();
@@ -155,7 +159,8 @@ public class EntityTrackerEntry {
                         this.v = 0;
                         // CraftBukkit start - Refresh list of who can see a player before sending teleport packet
                         if (this.tracker instanceof EntityPlayer) {
-                            this.scanPlayers(new java.util.ArrayList(this.trackedPlayers));
+                            // SportPaper - Fix invisibility on teleport
+                            this.scanPlayers(new ArrayList<EntityHuman>(this.tracker.world.players));
                         }
                         // CraftBukkit end
                         object = new PacketPlayOutEntityTeleport(this.tracker.getId(), i, j, k, (byte) l, (byte) i1, this.tracker.onGround);
-- 
2.30.2

