From 29cfa19f306abddf93012e89daa86af81eee78c7 Mon Sep 17 00:00:00 2001
From: linsaftw <linsaftw@users.noreply.github.com>
Date: Tue, 27 Apr 2021 09:53:13 -0300
Subject: [PATCH] Catch FastUtil exception


diff --git a/src/main/java/net/minecraft/server/ChunkProviderServer.java b/src/main/java/net/minecraft/server/ChunkProviderServer.java
index e58c022b..66b3b5e4 100644
--- a/src/main/java/net/minecraft/server/ChunkProviderServer.java
+++ b/src/main/java/net/minecraft/server/ChunkProviderServer.java
@@ -163,7 +163,12 @@ public class ChunkProviderServer implements IChunkProvider {
             chunk = originalGetChunkAt(i, j);
         }
 
-        unloadQueue.remove(LongHash.toLong(i, j)); // SportPaper
+        try {
+            unloadQueue.remove(LongHash.toLong(i, j)); // SportPaper
+        } catch (Exception e) {
+            // SportPaper - Catch possible NPEs (Concurrent modification?)
+        }
+        
         // If we didn't load the chunk async and have a callback run it now
         if (runnable != null) {
             runnable.run();
@@ -414,7 +419,11 @@ public class ChunkProviderServer implements IChunkProvider {
             this.saveChunkNOP(chunk);
             this.chunks.remove(chunk.chunkKey); // CraftBukkit
             if (!auto && this.unloadQueue.contains(chunk.chunkKey)) {
-                this.unloadQueue.remove(chunk.chunkKey);
+                try {
+                    this.unloadQueue.remove(chunk.chunkKey);
+                } catch (Exception e) {
+                    // SportPaper - Catch possible NPEs (Concurrent modification?)
+                }
             }
 
             // Update neighbor counts
@@ -442,12 +451,16 @@ public class ChunkProviderServer implements IChunkProvider {
             // SportPaper start
             LongIterator iterator = unloadQueue.iterator();
             for (int i = 0; i < 100 && iterator.hasNext(); ++i) {
-                long chunkcoordinates = iterator.nextLong();
-                iterator.remove();
-                // SportPaper end
-                Chunk chunk = this.chunks.get(chunkcoordinates);
-                if (chunk == null) continue;
-                unloadChunk(chunk, true); // SportPaper - Move to own method
+                try {
+                    long chunkcoordinates = iterator.nextLong();
+                    iterator.remove();
+                    // SportPaper end
+                    Chunk chunk = this.chunks.get(chunkcoordinates);
+                    if (chunk == null) continue;
+                    unloadChunk(chunk, true); // SportPaper - Move to own method
+                } catch (Exception e) {
+                    // SportPaper - Catch possible NPEs (Concurrent modification?)
+                }
             }
             // CraftBukkit end
 
-- 
2.31.1

