From a7addc1b58400190e483e0bc0cf4c02c2eccfff5 Mon Sep 17 00:00:00 2001
From: William Jeffcock <jeffcockw@gmail.com>
Date: Thu, 5 Aug 2021 19:17:47 +0100
Subject: [PATCH] Remove the world before nullifying chunkLoader &
 chunkProvider


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 1a92092b..0f75111e 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -912,6 +912,9 @@ public final class CraftServer implements Server {
             return false;
         }
 
+        worlds.remove(world.getName().toLowerCase());
+        console.worlds.remove(console.worlds.indexOf(handle));
+
         if (save) {
             try {
                 handle.save(true, null);
@@ -937,9 +940,6 @@ public final class CraftServer implements Server {
             chunkProviderServer.chunks.clear();
         }
 
-        worlds.remove(world.getName().toLowerCase());
-        console.worlds.remove(console.worlds.indexOf(handle));
-
         File parentFolder = world.getWorldFolder().getAbsoluteFile();
 
         // Synchronized because access to RegionFileCache.a is guarded by this lock.
-- 
2.17.1

