From d9173e8593b261c946686ac79bd9a3e08f4b7ea0 Mon Sep 17 00:00:00 2001
From: William Jeffcock <jeffcockw@gmail.com>
Date: Sat, 28 Aug 2021 23:32:33 +0100
Subject: [PATCH] Fix getPlayer not respecting viewer when looking for an exact
 player


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 0f75111e..6453951d 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -360,7 +360,7 @@ public final class CraftServer implements Server {
     public Player getPlayer(final String name, final CommandSender viewer) {
         Validate.notNull(name, "Name cannot be null");
 
-        Player found = getPlayerExact(name);
+        Player found = getPlayerExact(name, viewer);
         // Try for an exact match first.
         if (found != null) {
             return found;
-- 
2.17.1

