From f7980a6f0b2854a0a205ffb9d75d1f869487c758 Mon Sep 17 00:00:00 2001
From: Pugzy <pugzy@mail.com>
Date: Fri, 29 Sep 2023 17:35:55 +0100
Subject: [PATCH] Fix issues with non-interacting players


diff --git a/src/main/java/net/minecraft/server/IEntitySelector.java b/src/main/java/net/minecraft/server/IEntitySelector.java
index c27709e1..fb5bb04b 100644
--- a/src/main/java/net/minecraft/server/IEntitySelector.java
+++ b/src/main/java/net/minecraft/server/IEntitySelector.java
@@ -50,7 +50,7 @@ public final class IEntitySelector {
         }
 
         public boolean a(Entity entity) {
-            if (!entity.isAlive()) {
+            if (!entity.ad()) { // SportPaper - exempt non-interacting players
                 return false;
             } else if (!(entity instanceof EntityLiving)) {
                 return false;
diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index 3e5332f9..394b35b1 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -1289,6 +1289,7 @@ public abstract class World implements IBlockAccess {
 
         for (int j2 = 0; j2 < list.size(); ++j2) {
             if (entity.passenger != list && entity.vehicle != list) {
+                if (!((Entity) list.get(j2)).ad()) continue; // SportPaper - Fix boat placing collision
                 AxisAlignedBB axisalignedbb1 = ((Entity) list.get(j2)).S();
 
                 if (axisalignedbb1 != null && axisalignedbb1.b(axisalignedbb)) {
-- 
2.51.0

