From d6d18e96f0707b8036baa9a4b6fa6d7518bcd116 Mon Sep 17 00:00:00 2001
From: "BT (calcastor/mame)" <43831917+calcastor@users.noreply.github.com>
Date: Sun, 31 Dec 2023 22:03:05 -0800
Subject: [PATCH] Make dimension field non-final


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 9a2508c8..e536ae29 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -53,7 +53,7 @@ public class WorldServer extends World implements IAsyncTaskHandler {
     private List<NextTickListEntry> V = Lists.newArrayList();
 
     // CraftBukkit start
-    public final int dimension;
+    public int dimension; // SportPaper - make non-final
 
     // Add env and gen to constructor
     public WorldServer(MinecraftServer minecraftserver, IDataManager idatamanager, WorldData worlddata, int i, MethodProfiler methodprofiler, org.bukkit.World.Environment env, org.bukkit.generator.ChunkGenerator gen) {
-- 
2.43.0

